### PR TITLE
Create the Payment when the PayTomorrow order is prepared

### PR DESCRIPTION
**Pull request by abhishekgupta5**
**Issue:** https://linear.app/nebulab-retainers/issue/ABU-16/create-the-payment-when-the-paytomorrow-order-is-prepared

- [x] QAed in sandbox

### CODE REVIEW

**Comment by abhishekgupta5** [comment]
Payment object in checkout state -
image

File: app/services/solidus_pay_tomorrow/base_service.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module SolidusPayTomorrow
+  class BaseService
+    def initialize(**_args) end
```
Comment (DanielePalombo) [inline app/services/solidus_pay_tomorrow/base_service.rb:5]: Considering you added BaseService, what do you think to let to Client::BaseService to inherit from this Object?

ref https://github.com/nebulab/solidus_pay_tomorrow/blob/301245cdf773310a08d8d373683b644f3a5da76c/app/services/solidus_pay_tomorrow/client/base_service.rb#L7-L24

File: app/services/solidus_pay_tomorrow/handle_order_creation_service.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    def call
+      @result = create_order
+      create_payment
+      @result
+    end
+
+    private
+
+    def create_order
```
Comment (DanielePalombo) [inline app/services/solidus_pay_tomorrow/handle_order_creation_service.rb]: NIT: I think that create_order_on_pay_tomorrow will be a better name here

**Review by DanielePalombo — APPROVED** [review]
The PR works for me.
I just left some suggestions to improve the PR.

File: app/services/solidus_pay_tomorrow/base_service.rb (hunk 1 above)
Comment (abhishekgupta5) [reply app/services/solidus_pay_tomorrow/base_service.rb:5]: Yes, was thinking of doing it!
Will do before merging.

File: app/services/solidus_pay_tomorrow/handle_order_creation_service.rb (hunk 2 above)
Comment (abhishekgupta5) [reply app/services/solidus_pay_tomorrow/handle_order_creation_service.rb]: Done in de3d582fc3ea85827161f935a1388d6ae6f5ade4

File: app/services/solidus_pay_tomorrow/base_service.rb (hunk 1 above)
Comment (abhishekgupta5) [reply app/services/solidus_pay_tomorrow/base_service.rb:5]: Done in 272ffaf5f9a4cd678157d668abab6f88587d85a1

**Review by DanielePalombo — APPROVED** [review]
LGTM

**Pull request merged**